File: fleet-common/src/main/java/dev/odes/fleet/common/utils/CaseFormatUtils.java
package dev.odes.fleet.common.utils;

import com.google.common.base.CaseFormat;

public class CaseFormatUtils {

    // snake_case
    // lower_underscore
    public static String LOWER_UNDERSCORE(String word) {
        String str = "";
        CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, word);
        return str;
    }

    // UpperCamelCase
    // UpperCamel
    public static String UPPER_CAMEL(String word) {
        String str = "";
        CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, word);
        return str;
    }

    // UPPER_UNDERSCORE
    public static String UPPER_UNDERSCORE(String word) {
        String str = "";
        CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, word);
        return str;
    }

    // lowerCamelCase
    // lowerCamel
    public static String LOWER_CAMEL(String word) {
        String str = "";
        CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_CAMEL, word);
        return str;
    }


}

File: fleet-tool/src/utils/case_format_utils.js
import * as changeCase from "change-case";

// snake_case
// lower_underscore
export function lowerUnderscore(word) {
  return changeCase.snakeCase(word);
}

// UpperCamelCase
// UpperCamel
export function upperCamel(word) {
    return changeCase.pascalCase(word);
}

// UPPER_UNDERSCORE
export function UPPER_UNDERSCORE(word) {
    return changeCase.constantCase(word);
}

// lowerCamelCase
// lowerCamel
export function lowerCamel(word) {
    return changeCase.camelCase(word);
}
